earn/src/components/portfolio/modal/WithdrawModal.tsx

```diff
@@ -7,254 +7,136 @@ import { BaseMaxButton } from 'shared/lib/components/common/Input';
-
-  const { data: requestedShares, isLoading: convertToSharesIsLoading } = useContractRead({
-    address: kitty.address,
-    abi: KittyABI,
-    functionName: 'convertToShares',
-    args: [withdrawAmount.toString(GNFormat.INT)],
-    chainId: activeChain.id,
-  }) as { data: BigNumber | undefined; isLoading: boolean };
-
-  const gnRequestedShares = GN.fromBigNumber(requestedShares ?? BigNumber.from('0'), token.decimals);
-  // Being extra careful here to make sure we don't withdraw more than the user has
-  const numberOfSharesToRedeem = GN.min(gnRequestedShares, maxRedeemBalance);
-
-  const { config: redeemConfig } = usePrepareContractWrite({
-    address: kitty.address,
-    abi: KittyABI,
-    functionName: 'redeem',
-    args: [numberOfSharesToRedeem.toString(GNFormat.INT), accountAddress, accountAddress],
-    chainId: activeChain.id,
-    enabled: !numberOfSharesToRedeem.isZero() && !isPending,
-  });
-  const redeemUpdatedRequest = useMemo(() => {
-    if (redeemConfig.request) {
-      return {
-        ...redeemConfig.request,
-        gasLimit: redeemConfig.request.gasLimit.mul(GAS_ESTIMATE_WIGGLE_ROOM).div(100),
-      };
-    }
-    return undefined;
-  }, [redeemConfig.request]);
-  const {
-    write: contractWrite,
-    isSuccess: contractDidSucceed,
-    isLoading: contractIsLoading,
-    data: contractData,
-  } = useContractWrite({
-    ...redeemConfig,
-    request: redeemUpdatedRequest,
-  });
-
-  useEffect(() => {
-    if (contractDidSucceed && contractData) {
-      setPendingTxn(contractData);
-      setIsPending(false);
-      setIsOpen(false);
-    } else if (!contractIsLoading && !contractDidSucceed) {
-      setIsPending(false);
-    }
-  }, [contractDidSucceed, contractData, contractIsLoading, setPendingTxn, setIsOpen]);
-
-  let confirmButtonState = ConfirmButtonState.READY;
-
-  if (withdrawAmount.gt(maxWithdrawBalance)) {
-    confirmButtonState = ConfirmButtonState.INSUFFICIENT_ASSET;
-  } else if (isPending || convertToSharesIsLoading) {
-    confirmButtonState = ConfirmButtonState.PENDING;
-  } else if (numberOfSharesToRedeem.isZero()) {
-    confirmButtonState = ConfirmButtonState.LOADING;
-  }
-
-  const confirmButton = getConfirmButton(confirmButtonState, token);
-
-  function handleClickConfirm() {
-    if (confirmButtonState === ConfirmButtonState.READY && requestedShares) {
-      setIsPending(true);
-      contractWrite?.();
-    }
-  }
-
-  const isDepositAmountValid = withdrawAmount.isGtZero();
-  const shouldConfirmButtonBeDisabled = !(confirmButton.enabled && isDepositAmountValid);
+const Q112 = BigNumber.from('0x1000000000000000000000000000000');
 
-  return (
-    <FilledStylizedButton
-      size='M'
-      onClick={() => handleClickConfirm()}
-      fillWidth={true}
-      disabled={shouldConfirmButtonBeDisabled}
-    >
-      {confirmButton.text}
-    </FilledStylizedButton>
-  );
+function doesContainToken(token: Token, pair: LendingPair) {
```

Comment:
I feel this should be called something like `doesLendingPairContainToken`, and the first arg should be the pair and the second the token.